### DISCUSSION TITLE
Header recognition is incomplete,table with border in pdf

### DISCUSSION THREAD

**Discussion by stevenwu2017** [jsvine/pdfplumber#1177] [Ask for help with specific PDFs]
I had a problem with incomplete header recognition, table with border in pdf.
but i don't know what causes it. pdf file and debug table find result as follows:

[02498_ZH.pdf](https://github.com/user-attachments/files/16400686/02498_ZH.pdf)
![image_01](https://github.com/user-attachments/assets/9d0df228-4cf5-4d5e-ab80-38a8ca974339)
![image_02](https://github.com/user-attachments/assets/b330bb9d-ad25-4166-92ff-821863f7e54e)

python code:
import pdfplumber
    pdf_files = '02498_ZH.pdf'
    with pdfplumber.open(pdf_files) as pdf:
        # 遍历每个页面
        for idx, page in enumerate(pdf.pages):
            # if idx==0:
            deduped_page = page.dedupe_chars()
            tables = page.find_tables()
            for table in tables:
                content=table.extract()
                print(content)
                sub_page=page.crop(table.bbox)
                im = sub_page.to_image(resolution=200)
                im.debug_tablefinder().show()

**Comment by jsvine** [comment]
It seems that there are two issues:

- Some "invisible" rectangles in the headers
- The text in the headers are represented as graphical elements (curves, some of which at 90° angles) *in addition to* font-rendered text

Filtering out those extra graphical elements seems to work:

```python
def check(obj):
    # Keep only `chars` and objects with a black fill
    return obj.get("text") or obj.get("non_stroking_color") == (0, 0, 0)

page.filter(check).to_image().debug_tablefinder()
```

![image](https://github.com/user-attachments/assets/a9b6c80a-298b-4a8d-9488-11ce83b23c6d)

```python
page.filter(check).extract_tables()
```

```
[[['名稱', '於上市時需禁\n售的股票數目', '於上市時需禁售的\n股票數目之百分比\n(假設沒行使超額\n配售權)', '禁售期最後一天'],
  ['廣祺瑞騰股權投資合夥企業(有\n限合夥)', '', '', ''],
  ['Excel Rise Holdings Limited\n/逸昇控股有限公司',
   '2,585,094',
   '0.58%',
   '2024年7月4 日'],
  ['Guangzhou Yuexiu Jinchan\nPhase IV\nInvestment Partnership\n(Limited Partnership) / 廣\n州越秀金蟬四期投資合夥企業\n(有限合夥)',
   '2,843,603',
   '0.63%',
   '2024年7月4 日'],
  ['Mirae Asset Alpha Growth\nFund OFC', '2,018,195', '0.45%', '2024年7月4 日'],
  ['Jurastone Tech Singularity\nI Ltd', '431,070', '0.10%', '2024年7月4 日'],
  ['HUITENG CO-STONE INVESTMENT\nLIMITED', '1,421,801', '0.32%', '2024年7月4 日'],
  ['China World Investment\nLimited', '568,720', '0.13%', '2024年7月4 日'],
  ['小計', '179,222,503', '39.90%', ''],
  ['以上的表格所示禁售期屆滿日期依據招股章程的信息披露。', None, None, None]],
 [['名稱', '於上市時需禁\n售的股票數目', '於上市時需禁售的股\n票數目之百分比(假設\n沒行使超額配售權)', '禁售期最後一天'],
  ['Shenzhen Nanshan Strategic\nEmerging Industries\nInvestment Co., Ltd. / 深圳\n市南山戰略新興產業 投資有限\n公司',
   '18,166,000',
   '4.04%',
   '2024年7月4 日'],
  ['小計', '18,166,000', '4.04%', ''],
  ['根據相關的上市規則/指引資料，禁售期最後一天於2024年7月4日 。', None, None, None]]]
```

**Reply by stevenwu2017** [reply]
OK,It's work. Thank you very much for your solution !!!
